android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt

```diff
@@ -23,19 +24,28 @@ import javax.inject.Inject
 class ListingDetailsPageViewModel @Inject constructor(
 	private val listingsApi: ListingsApi,
 	savedStateHandle: SavedStateHandle,
-	private val navigationService: INavigationService
+	private val navigationService: INavigationService,
+	private val locationService: ILocationService,
 ) : SubletrViewModel<ListingDetailsPageUiState>() {
 	val navHostController: NavHostController get() = navigationService.navHostController
 
 	private val listingIdStream: BehaviorSubject<Int> =
 		BehaviorSubject.createDefault(checkNotNull(savedStateHandle["listingId"]))
-	private val imageIdsStream: BehaviorSubject<List<String>> = BehaviorSubject.createDefault(emptyList())
+	private val imageIdsStream: BehaviorSubject<List<String>> =
+		BehaviorSubject.createDefault(emptyList())
 	val favouritedStream: BehaviorSubject<Boolean> = BehaviorSubject.createDefault(false)
-	private val isFetchingImagesStream: BehaviorSubject<Boolean> = BehaviorSubject.createDefault(false)
+	private val isFetchingImagesStream: BehaviorSubject<Boolean> =
+		BehaviorSubject.createDefault(false)
 	private val listingDetailsStream: Observable<Result<ListingDetails>> = listingIdStream.map {
 		runCatching {
 			runBlocking {
-				val listing = listingsApi.listingsDetails(it)
+				val location = locationService.getLocation()
+				val listing =
+					listingsApi.listingsDetails(
+						listingId = it,
+						longitude = location!!.longitude.toFloat(),
```

Comment:
Can we maybe not take down the entire app if there is no location?

Reply:
Lol, I can use the location of like  university of waterloo as default

Reply:
Yeah, or you can just not make the call if the location is missing